enhancements/network/ocp-on-smart-nics.md

```diff
@@ -0,0 +1,313 @@
+     that can be used to control and provision it independently of
+     its host.
+
+  3. The NIC implements SR-IOV in such a way that we can connect
+     pods on the host directly to an OVN bridge on the NIC.
+
+  4. All of the above can be done without running any proprietary
+     drivers or third-party software on either the host's or the NIC's
+     RHCOS installation.
+
+## Design Details
+
+### Clusters Will Contain a Mix of Smart and "Dumb" NICs
+
+OVN Kubernetes is not currently capable of mixing offloaded and
+non-offloaded networking on the same node, so every pod on a node
+containing a Smart NIC must have offloaded networking, which, in the
+current architecture, means it must explicitly _request_ offloaded
+networking. In particular, this means that most pods created by
+OpenShift operators cannot be deployed on a node with a Smart NIC, and
+so all masters and at least a few workers will need to have "dumb"
+NICs.
+
+(While the exact architectural constraints may change in the future,
+it is likely that some customers will always want the possibility of
+only having Smart NICs on a subset of nodes.)
+
+### Overall Cluster Architecture
+
+There are several possible overall architectures:
+
+  - The Smart NICs could be nodes within the same OCP cluster as their
+    hosts. We would need to use some combination of labels and taints
+    to distinguish "real" nodes from Smart NIC nodes. (Among other
+    things, the Smart NICs would be their own `MachineConfigPool`,
+    separate from the masters and workers.) This may require changes
+    to tolerations and node selectors on existing DaemonSets, both
+    within OCP and in third-party operators or customer workloads (and
+    it may turn out that this approach is unworkable for this reason).
+
+  - The Smart NICs could be part of a second cluster running in
+    parallel with the primary cluster, with the control plane of the
+    NIC cluster running as pods within the primary cluster.
+
+  - The Smart NICs could be part of a second cluster which is
+    completely separate from the primary cluster, with its own master
+    nodes. (In some external discussion of this approach, it was
+    assumed that the masters in this scenario would have to be on ARM
+    hosts, but this isn't required; you can have a mix of x86_64
+    master nodes and ARM Smart NIC "worker" nodes.)
+
+  - The Smart NICs could be part of a second cluster, where the
+    control planes for both clusters are running in a different
+    cluster, ROKS-style.
+
+The latter two approaches allow for more isolation between the hosts
+and the NICs, since potentially even the cluster administrators of the
+primary cluster might not be able to influence the NIC cluster. This
+would be especially important if additional primary-cluster security
+and monitoring functions were offloaded to the NICs in the future.
+
+### Initial Development With Beta BlueField-2 Hardware
```

Comment:
My understanding of the BF-2 cards is that will be GA Hardware with a BMC network port and GA Hardware without a BMC network port (this was from a NVIDIA engineer working on the tiger team). Someone can correct me if I'm wrong about this. So if this is true, we as Red Hat need to decide if will support both versions of hardware. Or support the hardware with BMC network port upfront and add support for non-BMC hardware later.